timur/lib/client/jsx/components/query/query_results.tsx

```diff
@@ -39,6 +40,28 @@ const useStyles = makeStyles((theme) => ({
   }
 }));
 
+// https://reactjs.org/docs/error-boundaries.html
+// For #768, because just resetting the selection doesn't
+//    happen early enough in the CodeMirror lifecycle...?
+class ErrorBoundary extends React.Component {
```

Comment:
Yeah, also had to wrap the `<CodeMirror />` component in an `<ErrorBoundary />`, because it seemed like the `selection` updates just weren't happening quickly enough in the component lifecycle to avoid the error state. It's not a great solution, but I can't find any "real" API hooks that occur early enough in the CodeMirror lifecycle to handle this bug.